### ISSUE TITLE
[Improvement] Modifying the Locale system.

### ISSUE THREAD

**Issue by darbyjack** [aikar/commands#173]
What was the confusing thing is the world came down to a simple lack of documentation and understanding of how something worked. 

When using {@@} in description and message, you CAN'T use ROOT values. You MUST supply a children section for the languages. The improvement would be to allow multi-nested values along with root values in the @@

**Comment by SirYwell** [comment]
As I tried to use a custom yaml file for mesages right now, I came a across a problem with the same root. Currently, only the direct child nodes of root nodes are considered as candidates for message key. Example:
```yml
top:
  my.custom.value: MCV
  value: B
  second:
    value: C
    third:
      value: D
value: A
```
With this, `top.my.custom.value` and `top.value` are the only valid keys. I don't know if it is wanted, but personally I would prefer to have nested keys instead of period-separated keys (or similar). If you'd like that too, I can create a PR.

One additional point: Currently, only strings are accepted as values. `value: 1` won't produce the same result as `value: '1'`. That could be changed too if desired.
